### PR TITLE
Bottom navigation & named routes

### PR DESCRIPTION
**Pull request by Jei**
I created a HomePage that contains the top pages of the authenticated navigation flow. It uses a BottomNavigationBar to switch between the ConversationPage and the OptionsPage. All the other routes are reachable by using their `routeName` attribute together with the `Navigator.pushNamed` method.

I wanted to switch between SplashPage, AuthPage and HomePage using `Navigator.replaceNamed`, but I couldn't get it to work, and at this point, I'm not even sure it is necessary, so I left the signin/signout flow mostly as it was before.

I tried the changes on Android and they seem to work. Could you test them on iOS?

### CODE REVIEW

**Review by gaslitbytech — APPROVED** [review]
I can confirm that this works for iPhone. Thanks @Jei 

I think I'm getting the hang of it a bit more after watching some video's last night.

File: lib/home.dart [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import 'package:flutter/material.dart';
+import 'package:meetup_chatapp/conversations_page.dart';
+import 'package:meetup_chatapp/options_page.dart';
+
+class HomePage extends StatefulWidget {
```
Comment (nickmeinhold) [inline lib/home.dart]: What do you think about naming the file home_page.dart?  So far we seem to have fallen into that naming convention for other page widgets.

**Review by nickmeinhold — APPROVED** [review]
Looks great @Jei!  

We spoke about the navigator 2.0 coming through right?  It seems it's arrived - some breaking changes have landed in master, have you seen any of the announcements?  

https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/flutter-announce/DSAeObWGpoY/shAf76A5CQAJ

I only glanced through the announcement but didn't see any breaking changes that applied to this PR, but just wanted to point it out in case you hadn't seen it, and in case you were interested.

**Comment by nickmeinhold** [comment]
Just saw there is an official migration guide: https://flutter.dev/docs/release/breaking-changes/route-navigator-refactoring

Again, I haven't spent any time figuring out if there's anything relevant so sorry if there's not, just thought it was worth posting in case it's of interest.  

Thanks!

File: lib/home.dart (hunk 1 above)
Comment (Jei) [reply lib/home.dart]: Whoops, I think I got distracted. I'll fix it (and the splash page's name too) in another PR.

**Pull request merged**